### DISCUSSION TITLE
Why I receive PR_END_OF_FILE_ERROR/ERR_CONNECTION_TIMED_OUT while trying to access sites which is supposed to be accessed directly?

### DISCUSSION THREAD

**Discussion by Blabter** [v2rayA/v2rayA#1588] [Q&A]
1. I installed v2raya 2.2.5.7 in OpenWRT.
2. Configured VLESS server.
3. In "Settings -> Transparent Proxy/System Proxy" choose "On: Traffic splitting mode is the same as the Rule Port".
4. In "Settings -> Traffic splitting mode of Rule Port -> Configure" typed "default: direct" (So all sites must be accessed directly. No sites must be accessed through VPN)
5. Clicked "Connect" on my newly created server and clicked "Start" in top-left corner.

At first everything works as intended, all sites are accessed directly. But after some time I start to receive PR_END_OF_FILE_ERROR errors in firefox and ERR_CONNECTION_TIMED_OUT errors in chromium. 

Any ideas what causes this error?

I know that while using VLESS such errors can be caused by SNI site not responding to my queries, but since I configured v2ray not to route any queries through VPN I expect that any problems with VPN server will not cause problems on client side.

UPD: Seems like there is some problem with DNS resolution
```
$ ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=108 time=43.6 ms
^C
--- 8.8.8.8 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 43.617/43.617/43.617/0.000 ms
$ ping google.com
ping: google.com: Temporary failure in name resolution
```

**Comment by Sandr0oo** [comment]
I have the same problem. Help me, please.

**Reply by brys0** [reply]
I am also encountering this issue. Proxy worked fine on phone with v2rayng, but v2rayA does not work. The http test complains about the network being NOT stable. Any proxy option I set doesn't work

**Comment by Sandr0oo** [comment]
How to solve this problem?

**Reply by evstarshov** [reply]
I replaced default v2ray-core to xray-core on my ubuntu server and seems everything is work for me. 
Here some instructions - 
https://v2raya.org/docs/manual/use-other-core/ 
https://v2raya.org/docs/manual/variable-argument/#如何设置 
if pages doesn't open, change language to Chinese 
Seems there is no english version

**Reply by Sandr0oo** [reply]
Thank you! That seems to have helped!

**Reply by Blabter** [reply]
> I replaced default v2ray-core to xray-core on my ubuntu server and seems everything is work for me. Here some instructions - https://v2raya.org/docs/manual/use-other-core/ [https://v2raya.org/docs/manual/variable-argument/#如何设置](https://v2raya.org/docs/manual/variable-argument/#%E5%A6%82%E4%BD%95%E8%AE%BE%E7%BD%AE) if pages doesn't open, change language to Chinese Seems there is no english version

I replaced v2ray-core with xray-core, but problem still persists.

**Comment by LyraHarring** [comment]
I was getting [PR_END_OF_FILE_ERROR](https://cheapsslweb.com/blog/pr-end-of-file-error-in-firefox/) in Firefox and ERR_CONNECTION_TIMED_OUT in Chromium after some time, even though I set default: direct in v2raya. At first, everything worked fine, then the errors started randomly. What fixed it for me was replacing the default v2ray-core with xray-core on my server. It is a specific issue in Firefox when the browser cannot establish a secure connection with a website. This problem occurs on the client side and is associated with cipher suites and instructions for securing and verifying network connections using cryptographic protocols such as SSL and TLS. After switching to xray-core, the errors stopped completely. You can follow the steps here: https://v2raya.org/docs/manual/use-other-core/.

**Comment by Firestar-Reimu** [comment]
PR_END_OF_FILE_ERROR appears only on Firefox, other browser does not have this warning message

check if the website which says PR_END_OF_FILE_ERROR in Firefox can be connected via other chromium browsers

**Reply by Blabter** [reply]
>ERR_CONNECTION_TIMED_OUT errors in chromium.

**Comment by machsix** [comment]
If you are using `ws+cloudflare`, you need to set the SNI the same as your domain. Otherwise, you will get `PR_END_OF_FILE_ERROR`
```json
   {
      "tag": "proxy",
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "your_domain.com",
            "port": 443,
            "users": [
              {
                "id": "29",
                "alterId": 0,
                "security": "auto"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls",
        "tlsSettings": {
          "allowInsecure": true,
          "serverName": "your_domain.com",  !<----- this should be your domain, not baidu.com or other thing
          "show": false
        },
        "wsSettings": {
          "path": "/about",
          "headers": {
            "Host": "baidu.com"  !<--- doesn't matter
          }
        }
      },
      "mux": {
        "enabled": false,
        "concurrency": -1
      }
    },

```

**Reply by GTManR** [reply]
You are right;I used CloudFlare optimized IP,I think this is a bug

**Reply by Blabter** [reply]
I am not using WS or cloudflare

**Comment by tiger31** [comment]
Replacing v2ray package to xray solved this issue for me on NixOS (25.11 channel):
```nix
  # /etc/nixos/configuration.nix
  
  services.v2raya = {
    enable = true;
    cliPackage = pkgs.xray;
  };
```
